pkgs/development/compilers/gcc/common/checksum.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, nukeReferences
+, langC
+, langCC
+}:
+
+let
+  enableChecksum = (with stdenv; buildPlatform == hostPlatform && hostPlatform == targetPlatform) && langC && langCC;
+in
+(pkg: pkg.overrideAttrs (previousAttrs: lib.optionalAttrs enableChecksum {
+  outputs = previousAttrs.outputs ++ lib.optionals enableChecksum [ "checksum" ];
+  # This is a separate phase because gcc assembles its phase scripts
+  # in bash instead of nix (we should fix that).
+  preFixupPhases = (previousAttrs.preFixupPhases or []) ++ [ "postInstallSaveChecksumPhase" ];
+  #
+  # gcc uses an auxiliary utility `genchecksum` to md5-hash (most of) its
+  # `.o` and `.a` files prior to linking (in case the linker is
+  # nondeterministic).  Since we want to compare across gccs built from two
+  # separate derivations, we wrap `genchecksum` with a `nuke-references`
+  # call.  We also stash copies of the inputs to `genchecksum` in
+  # `$checksum/inputs/` -- this is extremely helpful for debugging since
+  # it's hard to get Nix to not delete the $NIX_BUILD_TOP of a successful
+  # build.
+  #
+  postInstallSaveChecksumPhase = ''
+    mv gcc/build/genchecksum gcc/build/.genchecksum-wrapped
+    cat > gcc/build/genchecksum <<\EOF
+    #!/bin/sh
```

Comment:
```suggestion
    #!${runtimeShell}
```

Reply:
This script never runs outside of a derivation, and the text above never winds up in a derivation output (unless I made a mistake).  Are you sure this should be `runtimeShell`?

I thought `runtimeShell` was when you needed a shell that would be invoked outside of a derivation, like writing a script into `$out`.  But maybe that's not correct.  I couldn't find an explanation of exactly when to use `runtimeShell` in the nixpkgs manual (`rg runtimeShell nixpkgs/doc`) and `all-packages.nix` doesn't have a comment near that attribute.

Reply:
runtimeShell is the interpreter used to run bash scripts. Whether it's in our outside a drv doesn't matter.

Actually, I wouldn't even expect `/bin/sh` to be present in a drv. It's always better to depend on paths directly.